### PR TITLE
fix(deps): allow PHP 8.4

### PR DESCRIPTION
**Pull request by ben-challis**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -88,7 +89,7 @@ jobs:
         run: "composer code-style:check"
 
       - name: "Run PHPStan"
-        if: ${{ matrix.dependencies == 'locked' }}
+        if: ${{ matrix.dependencies == 'locked' && matrix.php-version == '8.2' }}
```
Comment (martin-georgiev) [inline .github/workflows/ci.yml:92]:
```suggestion
        if: ${{ matrix.dependencies == 'locked' && matrix.php-version == '8.3' }}
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -88,7 +89,7 @@ jobs:
         run: "composer code-style:check"
 
       - name: "Run PHPStan"
-        if: ${{ matrix.dependencies == 'locked' }}
+        if: ${{ matrix.dependencies == 'locked' && matrix.php-version == '8.3' }}
```
Comment (ben-challis) [inline .github/workflows/ci.yml:92]:
```suggestion
        if: ${{ matrix.dependencies == 'locked' && matrix.php-version == '8.2' }}
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -22,6 +22,7 @@ jobs:
         php-version:
           - "8.2"
           - "8.3"
+          - "8.4"
```
Comment (marmichalski) [inline .github/workflows/ci.yml:25]: Can we also ensure PHP 8.4 is actually allowed in composer.json: https://github.com/Lendable/composer-license-checker/blob/967f594e79e09ce49ca7593cfe2dada8d652b5ad/composer.json#L13

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ben-challis) [inline .github/workflows/ci.yml:92]:
```suggestion
        if: ${{ matrix.dependencies == 'locked' }}
```

**Review by marmichalski — APPROVED** [review]

**Comment by ben-challis** [comment]
Going with #394 instead

**Pull request closed without merging**